src/test/java/io/kestra/plugin/serdes/avro/infer/InferAvroSchemaTest.java

```diff
@@ -0,0 +1,133 @@
+package io.kestra.plugin.serdes.avro.infer;
+
+import com.google.common.collect.ImmutableMap;
+import io.kestra.core.junit.annotations.KestraTest;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.runners.RunContextFactory;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.storages.StorageInterface;
+import io.kestra.core.utils.IdUtils;
+import io.kestra.core.utils.TestsUtils;
+import io.kestra.plugin.serdes.avro.InferAvroSchemaFromIon;
+import io.kestra.plugin.serdes.csv.IonToCsv;
+import jakarta.inject.Inject;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.time.*;
+import java.util.Date;
+import java.util.List;
+
+import static io.kestra.core.utils.Rethrow.throwConsumer;
+
+@KestraTest
+public class InferAvroSchemaTest {
+    @Inject
+    StorageInterface storageInterface;
+
+    @Inject
+    RunContextFactory runContextFactory;
+
+    @Test
+    void ion() throws Exception {
+        var tempFile = File.createTempFile(InferAvroSchemaTest.class.getSimpleName().toLowerCase() + "_", ".ion");
+        try (var output = new FileOutputStream(tempFile)) {
+            List.of(
+                    ImmutableMap.builder()
```

Comment:
Personal opinion, but I try to avoid using google common when it's not necessary. avoiding to depends on an external lib

Doing the following, do the trick ;) 
```
new HashMap<>() {{
  put("myInt", 2);
  put("myFloat", 3.2F);
}}
```